### PR TITLE
feat: search and select for workspaces in sidebar

### LINKED ISSUE #3167: Feature: Make workspaces dropdown searchable
As a user of many workspaces, I want the workspaces dropdown in the sidebar to be searchable, so that I can drill down to a specific workspace quicker.

### PR DESCRIPTION
**Pull request by zeucapua**
## Description

Allows users to search workspaces from the sidebar. Required a refactor of `SingleSelect` to use `Popover` and `Command` components, similar to `MultiSelect`.

## Related Tickets & Documents

Closes #3167 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/20603494/417b440c-2b88-4241-bd4b-223399c24aae

## Steps to QA

1. Use the sidebar workspaces dropdown
2. Try searching, using arrow keys, clicking to select workspace options
3. Ensure it redirects to the workspace

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [x] Tier 3
- [ ] Tier 4

## [optional] What gif best describes this PR or how it makes you feel?

search-shanique-drummond-gif

### CODE REVIEW

File: components/shared/DayRangePicker.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ export const DayRangePicker = ({ onDayRangeChanged }: DayRangePickerProps) => {
 
   return (
     <label className="w-max font-semibold">
-      <span className="sr-only">Period:</span>
+      <span className="sr-only">Range:</span>
```
Comment (nickytonline) [inline components/shared/DayRangePicker.tsx:44]: This is so assistive technologies can read the label. As far as I know, the `insetLabel` is `::before` content which won't get picked up by assistive technologies.

File: components/atoms/Select/single-select.tsx [hunk 2]
```diff
@@ -32,9 +31,7 @@ const SingleSelect = ({
   const [inputValue, setInputValue] = useState("");
   const [isOpen, setIsOpen] = useState(false);
 
-  const filteredOptions = options.filter((option) => option.label.toLowerCase().includes(inputValue.toLowerCase()));
-
-  const items = isSearchable ? filteredOptions : options;
+  const current = options.find((option) => option.value === value ?? "");
```
Comment (nickytonline) [inline components/atoms/Select/single-select.tsx:34]: Is the empty string required? If the value is undefined, it should result in false.

```suggestion
  const current = options.find((option) => option.value === value);
```

Comment (zeucapua) [reply components/atoms/Select/single-select.tsx:34]: Oh you're correct, I was working on autopilot lol

**Review by nickytonline — APPROVED** [review]
:shipit:

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**